### PR TITLE
[ts sdk] Add new tx.object methods

### PR DESCRIPTION
**Pull request by hayes-mysten**
## Description 

Describe the changes or additions included in this PR.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: sdk/typescript/test/e2e/object-cache.test.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { CachingTransactionExecutor } from '../../src/transactions/executor/cach
 import { normalizeSuiAddress } from '../../src/utils';
 import { setup, TestToolbox } from './utils/setup';
 
-describe('CachingTransactionExecutor', async () => {
+describe('CachingTransactionExecutor', { retry: 3 }, async () => {
```
Comment (hayes-mysten) [inline sdk/typescript/test/e2e/object-cache.test.ts:13]: these tests fail intermittently, with "transaction timed out before reaching finality" which I think happens at epoch boundaries, but has been kinda painful to try to debug.  I am pretty confident these issues are unrelated to the code bing tested, and more to do with how we are running the tests

File: sdk/deepbook/vitest.config.ts [hunk 2]
```diff
@@ -5,8 +5,8 @@ import { defineConfig } from 'vitest/config';
 
 export default defineConfig({
 	test: {
-		minThreads: 1,
```
Comment (hayes-mysten) [inline sdk/deepbook/vitest.config.ts:8]: These options don't exist anymore, I suspect that running too many tests in parallel is whats causing the execution timeouts and flakiness

**Review by pchrysochoidis — APPROVED** [review]

**Review by williamrobertson13 — APPROVED** [review]

**Review by Jordan-Mysten — APPROVED** [review]
